src/mame/sinclair/tsconf_m.cpp

```diff
@@ -716,21 +716,25 @@ u8 tsconf_state::tsconf_port_77_zctr_r(offs_t port)
 
 void tsconf_state::tsconf_port_57_zctr_w(offs_t port, u8 data)
 {
-	if (!m_zctl_cs)
+	if (m_zctl_cs)
+		return;
+
+	for (u8 m = 0x80; m; m >>= 1)
 	{
-		for (u8 m = 0x80; m; m >>= 1)
-		{
-			m_sdcard->spi_clock_w(CLEAR_LINE); // 0-S R
-			m_sdcard->spi_mosi_w(data & m ? 1 : 0);
-			m_sdcard->spi_clock_w(ASSERT_LINE); // 1-L W
-		}
+		m_sdcard->spi_clock_w(CLEAR_LINE); // 0-S R
+		m_sdcard->spi_mosi_w(data & m ? 1 : 0);
+		m_sdcard->spi_clock_w(ASSERT_LINE); // 1-L W
 	}
 }
 
 u8 tsconf_state::tsconf_port_57_zctr_r(offs_t port)
 {
+	if (m_zctl_cs)
+		return 0xff;
+
+	u8 data = m_zctl_di;
 	tsconf_port_57_zctr_w(0, 0xff);
-	return m_zctl_cs ? 0xff : m_zctl_di;
+	return data;
```

Comment:
You should probably check for `machine().side_effects_disabled()` before doing the write.  Otherwise having a debugger memory view open will constantly trigger this.

Reply:
In tsconf_port_57_zctr_w